iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -72,7 +72,11 @@ public void Shutdown()
                 Logging.Enter(this, _amqpIotConnection, nameof(Shutdown));
 
             _amqpAuthenticationRefresher?.StopLoop();
-            _amqpIotConnection?.SafeClose();
+            if (_amqpIotConnection != null)
```

Comment:
doing a if condition because the following is not allowed:
_amqpIotConnection?.Closed -= OnConnectionClosed;